doc/vhpidirect/examples/arrays.rst

```diff
@@ -0,0 +1,63 @@
+
+This example includes multiple solutions to set the bounding size once only (either in C or in VHDL), and have the
+parameter passed to the other language so that matching types are defined. Moreover, independently of where is the size
+defined, it is possible to allocate and free the buffers in any of the languages. The following examples show how to
+define them in C or VHDL and have them (de)allocated in either of them.
+
+.. ATTENTION::
+  Pointers/accesses MUST be freed/deallocated in the same language that was used to allocate them. Hence, it is not
+  possible to allocate in VHDL and free in C or *vice versa*.
+
+:cosimtree:`Sized in C <vhpidirect/arrays/intvector/csized>`
+------------------------------------------------------------
+
+Integer arrays fully defined in C can be passed to VHDL by first passing their size, so that an appropriate array
+type can be created in VHDL to hold the array's access (pointer). After that, another VHPIDIRECT subprogram can be
+defined to return the array access.
+
+This example shows how to hardcode both the length and the initial content of an array in C. Matching types are created in
+VHDL, the pointer is passed and the content is read and modified from VHDL.
+
+If the integer array must be created or filled at runtime by some more advanced process, it is possible to execute the GHDL
+simulation within a custom ``main()`` entrypoint (see :ref:`COSIM:VHPIDIRECT:Examples:wrapping:basic`). This use case is
+included in the example too. By using ``main.c`` instead of ``caux.c``, the content of the array is written programatically
+in C, before calling ``ghdl_main``. Note that the content of the array is read from C both before and after executing the
+simulation.
+
+.. NOTE::
+  There is no explicit example about how to have the size defined in C, but have the (de)allocation performed in VHDL.
+  However, implementing such a solution is a matter of combining these examples with the VHDL-sized ones below.
+
+:cosimtree:`Sized in VHDL <vhpidirect/arrays/intvector/vhdlsized>`
+------------------------------------------------------------------
+
+Complementing the examples above, when the size of a bounded/constrained array is defined in VHDL, it is possible to have
+the (de)allocation performed in either VHDL or C. However, while accesses to constrained VHDL types do contain metada about
+the bounds, pointers in C do not. Hence, in these examples, the length is explicitly passed along with the pointer/access.
+Note that other possible implementations would save the length in a variable in C, so that it does not need to be passed
+each time. This is done in the example with ``main()`` above.
+
+In this example two equivalent architectures are provided. In ``calloc`` allocation is done in C, while in ``vhdlalloc``
+the (de)allocation is done in VHDL. Apart from that, both are functionally equivalent:
```

Comment:
Could it be "(de)allocation is invokde from VHDL"... there is no VHDL equivalent of malloc/free, it is C side. Same on line 40.

Reply:
In calloc, both allocation and deallocation (free) are done in C, but invoked from VHDL. In vhdlalloc, both allocation (new) and deallocation are done in VHDL. Hence, I use allocate == new, and free == deallocate. Terms are equivalent, but the implementation is not. That's why, I believe, it is not possible to allocate in a language and deallocate in the other.

Anyway, I will reword it.